### PR TITLE
APMSPII-196 | Add address description

### PR DESCRIPTION
**Pull request by ben-clarke**
## Summary
* Add address description

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: specification/personal-demographics.yaml [hunk 1]
```diff
@@ -1057,26 +1057,63 @@ paths:
...
+        *	line 3 - dependent locality/village, locality (if present), e.g. `Boxgrove`
+        *	line 4 - post town, e.g. `Leeds`
+        *	line 5 - county (if present), e.g. `West Yorkshire`
 
-        When updating an address, lines 1 or 2, and line 4 must be populated; line 5 should not be included in vernacular addresses but may be included in PAF-derived addresses.
+        When updating an address, lines 1 or 2, and line 4 should be populated; line 5 should not be included in manually created addresses but may be included in PAF derived addresses.
+
+        Address lines should be populated otherwise the message may be rejected. There are exceptions:
+        * if a postcode and PAF key are submitted, in which case the lines will be automatically populated, however if there are no matches or too many matches the message will be rejected due to missing address lines
+        * a pseudo postcode is used. For example `ZZ99 3VZ` meaning `no fixed abode`
```
Comment (alisonkinloch-nhs) [inline specification/personal-demographics.yaml:1072]: Is there a link we can add in here to the list of peusdo postcodes

File: specification/personal-demographics.yaml [hunk 2]
```diff
@@ -1057,26 +1057,63 @@ paths:
...
-        *	Line 4 - post town, e.g. `Leeds`
-        *	Line 5 - county (if present), e.g. `West Yorkshire`.
+        *	line 1 - premises ID and/or house name, e.g. `Flat 1` or `The Old Schoolhouse`
+        *	line 2 - house number, dependent thoroughfare name and descriptor (if present), thoroughfare name and descriptor, e.g. `23 Mill Lane`
+        *	line 3 - dependent locality/village, locality (if present), e.g. `Boxgrove`
+        *	line 4 - post town, e.g. `Leeds`
+        *	line 5 - county (if present), e.g. `West Yorkshire`
 
-        When updating an address, lines 1 or 2, and line 4 must be populated; line 5 should not be included in vernacular addresses but may be included in PAF-derived addresses.
+        When updating an address, lines 1 or 2, and line 4 should be populated; line 5 should not be included in manually created addresses but may be included in PAF derived addresses.
```
Comment (alisonkinloch-nhs) [inline specification/personal-demographics.yaml:1068]: This isn't consistent with the advice below which says to exclude blank lines.  here we are saying populate lines 1,2,4 so 3 would be blank.  Below we say to populate as 1,2,3.    It is the overloading of the term lines, as line 4 does not correspond to address position 4 if 3 does not exist.

File: specification/personal-demographics.yaml [hunk 3]
```diff
@@ -1090,10 +1127,54 @@ paths:
...
         * any `billing` address must have both a `business effective from` and a `business effective to` date. The provision of a business effective to date has particular importance in order to avoid correspondence addresses that are no longer relevant to the patient still being held as current data available to any system retrieving the patient record. A suggested default where no actual business effective to is known is 30 days later than the business effective from, up to a maximum of 12 months.
+        * the date period is optional; where present they must be valid dates and the `end` date cannot be before the `start` date
+        * the business effective `start` date is optional, however if provided cannot be a future date. If it is not provided it will default to the date of update
```
Comment (alisonkinloch-nhs) [inline specification/personal-demographics.yaml:1135]: terminology mixed again here business effective is used but in the spec it is not in a field named as that.

File: specification/components/schemas/Address.yaml [hunk 4]
```diff
@@ -14,15 +14,59 @@ properties:
...
     description: |
       Purpose of this address:
-      * home - a communication address at a home.
-      * work - an office address. This can be returned due to legacy data but cannot be added or replaced on update.
-      * temp - a temporary address. The period can provide more detailed information.
-      * billing - a corresponding address nominated by the patient where communication can be sent.
+        * `home` - the home address is the patient's normal residence. Home address is also known as usual, main, registered, current or permanent address
+        * `temp` - a temporary address is an address used for a set period of time, but where the patient's home, permanent address remains unchanged
+        * `billing` - an address used for correspondence purposes only
+        * work - an office address. This can be returned due to legacy data but cannot be added or replaced on update
```
[the remark this answers is not in the document]
Comment (ben-clarke) [reply specification/components/schemas/Address.yaml:20]: Changed

File: specification/components/schemas/Address.yaml [hunk 5]
```diff
@@ -14,15 +14,59 @@ properties:
...
+        Where multiple instances already exist for the patient it is not expected that local systems should manage those, but should choose the most appropriate one to maintain (e.g. by examining business effective dates).
+
+        A `home` address is the patient's main residential address and should normally be used for all clinical and demographic purposes, including clinical and appointment correspondence.
+        However additionally, `temp` and `billing` addresses may be provided by a patient when there is a requirement to record an alternative location for the purposes of unbroken care.
+        When sending correspondence to a patient:
+        *	a present and valid `billing` address may take precedence over `home` and `temp` addresses. A patient should have only a single current `billing` address. An address is considered 'valid' according to its business effective from and to dates.
+        *	if no current `billing` address is provided, a `temp` address may take precedence over the `home` address, again if it is valid according to its business effective from and to dates.
```
[the remark this answers is not in the document]
Comment (ben-clarke) [reply specification/components/schemas/Address.yaml:30]: Change to period start and end

File: specification/components/schemas/Address.yaml [hunk 6]
```diff
@@ -14,15 +14,59 @@ properties:
...
+        * if there is no valid, current `billing` and/or `temp` address, the `home` address must be used.
     enum: [home, work, temp, billing]
     example: home
+  text:
+    type: string
+    description: |
+      Where a `temp` address is provided a descriptor text must be sent.
+      The list of possible values is:
```
[the remark this answers is not in the document]
Comment (ben-clarke) [reply specification/components/schemas/Address.yaml:38]: Changes

File: specification/personal-demographics.yaml (hunk 1 above)
Comment (ben-clarke) [reply specification/personal-demographics.yaml:1072]: Links in same document don't work in APIGEE

File: specification/personal-demographics.yaml (hunk 3 above)
Comment (ben-clarke) [reply specification/personal-demographics.yaml:1135]: Changed

**Pull request merged**